walk-in-interview/src/main/webapp/new-job/script.js

```diff
@@ -0,0 +1,296 @@
+
+  const datePicker =
+    document.getElementById('new-job-expiry');
+  datePicker.setAttribute('min', min);
+  datePicker.setAttribute('max', max);
+}
+
+/**
+ * Gets job detail from user input.
+ * @return {Object} containing the user inputs.
+ */
+function getJobDetailsFromUserInput() {
+  const name = document.getElementById('new-job-title').value;
+  const address =
+    document.getElementById('new-job-address').value;
+  const description =
+    document.getElementById('new-job-description').value;
+  const payFrequency =
+    document.getElementById('new-job-pay-frequency').value;
+  const payMin =
+    document.getElementById('new-job-pay-min').valueAsNumber;
+  const payMax =
+    document.getElementById('new-job-pay-max').valueAsNumber;
+
+  const requirementsCheckboxes =
+    document.getElementsByName(STRINGS['new-job-requirements-list']);
+  const requirementsList = [];
+  requirementsCheckboxes.forEach(({checked, id}) => {
+    if (checked) {
+      requirementsList.push(id);
+    }
+  });
+
+  const expiry =
+    document.getElementById('new-job-expiry')
+        .valueAsNumber;
+  const duration =
+    document.getElementById('new-job-duration').value;
+
+  const jobDetails = {
+    jobTitle: name,
+    jobLocation: {
+      address: address,
+      lat: 1.3039, // TODO(issue/13): get these from places api
+      lon: 103.8358,
+    },
+    jobDescription: description,
+    jobPay: {
+      frequency: payFrequency,
+      min: payMin,
+      max: payMax,
+    },
+    requirements: requirementsList,
+    postExpiry: expiry,
+  };
+
+  if (duration !== '') {
+    jobDetails.jobDuration = duration;
+  }
+
+  return jobDetails;
+}
+
+/**
+ * Validates the user input
+ * @return {boolean} depending on whether the input is valid or not.
+ */
+function validateRequiredUserInput() {
+  // TODO(issue/19): add more validation checks
+  const name = document.getElementById('new-job-title').value;
+  const address =
+    document.getElementById('new-job-address').value;
+  const description =
+    document.getElementById('new-job-description').value;
+  const payFrequency =
+    document.getElementById('new-job-pay-frequency').value;
+  const payMin =
+    document.getElementById('new-job-pay-min').valueAsNumber;
+  const payMax =
+    document.getElementById('new-job-pay-max').valueAsNumber;
+  const expiry =
+    document.getElementById('new-job-expiry')
+        .valueAsNumber;
+
+  if (payMin > payMax) {
+    return false;
+  }
+
+  if (name !== '' && address !== '' && description !== '' &&
+    payFrequency !== '' && payMin !== '' && payMax !== '' &&
+    !Number.isNaN(expiry)) {
+    return true;
+  }
+
+  return false;
+}
+
+const submitButton = document.getElementById('new-job-submit');
+submitButton.addEventListener('click', (_) => {
+  if (!validateRequiredUserInput()) {
+    document.getElementById('new-job-error-message').innerText =
+      STRINGS['new-job-error-message'].message;
```

Comment:
Maybe validateRequiredUserInput some how returns an error message indicating which field is missing/ wrongly entered? And we show it here? If no error message, the input is correct? 
Usually it's not trivial and friendly when a form just says "there's an error" without pointing out where or why.

Reply:
Agree